lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,18 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  def set_symlink_folder
+    puppetModulesVendorFolder = "#{ENV['ProgramFiles']}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
```

Comment:
Agreed on changing how we get ProgramFiles - in the old way we didn't have the accidental possibility of ending up on `C:\Program Files (x86)` but here we do. Just moving things in at the moment. Working on specs to start defining exactly how this will look.